tests/dogtag/pytest-ansible/pytest/performance_test/test_acme_cert_enrollment.py

```diff
@@ -0,0 +1,238 @@
+"""
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+#   Description: Load Test client Server Side configuration
+#   Steps :-
+#
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+#   Author: Deepak Punia <dpunia@redhat.com>
+#
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Copyright Red Hat, Inc.
+# SPDX-License-Identifier: GPL-2.0-or-later
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+"""
+
+import argparse
+import subprocess
+import json
+import os, sys
+import base64
+import binascii, time
+import hashlib
+import re, logging
+import threading
+import OpenSSL
+from acme import crypto_util
+from timeit import default_timer as timer
+from urllib.request import urlopen, Request
+
+DEFAULT_DIRECTORY_URL = "http://localhost:8080/acme/directory"
+
+LOGGER = logging.getLogger(__name__)
+LOGGER.addHandler(logging.StreamHandler())
+LOGGER.setLevel(logging.INFO)
+
+parser = argparse.ArgumentParser(formatter_class=argparse.RawDescriptionHelpFormatter)
+parser.add_argument("--domain", required=True, help="Client Domain name")
+parser.add_argument("--acme-dir", required=True, help="path to the .well-known/acme-challenge/ directory")
+parser.add_argument("--directory-url", required=True, default=DEFAULT_DIRECTORY_URL,
+                    help="certificate authority directory url")
+parser.add_argument("--number-of-threads", required=True, help="Number of threads", type=int)
+parser.add_argument("--number-of-tests-per-thread", required=True, help="Number of test per thread", type=int)
+
+args = parser.parse_args()
+LOGGER.setLevel(LOGGER.level)
+
+account_key = "{}.key".format(args.domain)
+csr = "{}.csr".format(args.domain)
+
+
+def new_csr_comp(domain_name, pkey_pem=None):
+    """Create certificate signing request."""
+    if pkey_pem is None:
```

Comment:
Rather than racing to create the private key, why not do this in `__main__` from a new method? `create_private_key()` or something?

Reply:
Edit: Never mind, misunderstood what this was doing. Please ignore. :-)